test/e2e/protocol/proposer.test.mjs

```diff
@@ -127,6 +127,37 @@ describe('Basic Proposer tests', () => {
     }
   });
 
+  it('should fail to register a proposer without a valid API key', async () => {
+    try {
+      bootProposer.setApiKey('test');
+      await bootProposer.registerProposer(testProposersUrl[2], minimumStake);
+      expect.fail();
+    } catch (err) {
+      expect(err);
+    } finally {
+      bootProposer.setApiKey(environment.AUTH_TOKEN);
+    }
+  });
+
+  it('should fail to register a proposer without an API key', async () => {
+    try {
+      bootProposer.resetApiKey();
+      await bootProposer.registerProposer(testProposersUrl[2], minimumStake);
+      expect.fail();
+    } catch (err) {
+      expect(err);
+    } finally {
+      bootProposer.setApiKey(environment.AUTH_TOKEN);
+    }
+  });
+
+  it('should access any public route with any API key', async () => {
+    bootProposer.resetApiKey();
+    await bootProposer.unprocessedTransactionCount();
+    bootProposer.setApiKey(environment.AUTH_TOKEN);
+    await bootProposer.unprocessedTransactionCount();
+  });
```

Comment:
Is this test to demonstrate that unprotected routes are available both with and without API key? If so I think these are actually 2 tests - one per scenario.

Specially since right/wrong API key seems that won't matter as the middleware is not present.

Also, I guess we are relying on Promises being resolved, but IMO we should try to write assertions. Here you could assert against status code `200` for example.